modules/migration/oaep-data-migration/src/main/java/org/wso2/migration/internal/MigrationServiceComponent.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.migration.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.base.api.ServerConfigurationService;
+import org.wso2.carbon.core.internal.CarbonCoreDataHolder;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.migration.DataMigrationClient;
+import org.wso2.migration.DataMigrationClientImpl;
+
+/**
+ * @scr.component name="org.wso2.carbon.migration.internal" immediate="true"
+ * @scr.reference name="realm.service"
+ * interface="org.wso2.carbon.user.core.service.RealmService" cardinality="1..1"
+ * policy="dynamic" bind="setRealmService" unbind="unsetRealmService"
+ * @scr.reference name="registry.service"
+ * interface="org.wso2.carbon.registry.core.service.RegistryService" cardinality="1..1"
+ * policy="dynamic" bind="setRegistryService" unbind="unsetRegistryService"
+ * @scr.reference name="serverconfiguration.service"
+ * interface="org.wso2.carbon.base.api.ServerConfigurationService" cardinality="1..1"
+ * policy="dynamic" bind="setServerConfigurationService" unbind="unsetServerConfigurationService"
+ */
+public class MigrationServiceComponent {
+    private static final Log LOG = LogFactory.getLog(MigrationServiceComponent.class);
+
+    /**
+     * Method to activate bundle.
+     *
+     * @param context OSGi component context.
+     */
+    protected void activate(ComponentContext context) {
+        try {
+            BundleContext bundleContext = context.getBundleContext();
+            bundleContext.registerService(DataMigrationClient.class, new DataMigrationClientImpl(), null);
+
+            // if -Dmigrate option is used.
+            String migrate = System.getProperty("migrate");
+            if (Boolean.parseBoolean(migrate)) {
+                LOG.info("Executing Migration client : " + DataMigrationClient.class.getName());
```

Comment:
executing or initializing?

Reply:
in here we are executing the client after initializing it..